tests/components/egps/conftest.py

```diff
@@ -0,0 +1,63 @@
+"""Configure tests for egps."""
+
+from collections.abc import Generator
+from typing import Final
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from pyegps.fakes.powerstrip import FakePowerStrip
+import pytest
+
+from homeassistant.components.egps.const import CONF_DEVICE_API_ID, DOMAIN
+from homeassistant.const import CONF_NAME
+
+from tests.common import MockConfigEntry
+
+DEMO_CONFIG_DATA: Final = {
+    CONF_NAME: "Unit Test",
+    CONF_DEVICE_API_ID: "DYPS:00:11:22",
+}
+
+
+@pytest.fixture
+def demo_config_data() -> dict:
+    """Return valid user input."""
+    return {CONF_DEVICE_API_ID: DEMO_CONFIG_DATA[CONF_DEVICE_API_ID]}
+
+
+@pytest.fixture
+def valid_config_entry() -> MockConfigEntry:
+    """Return a valid egps config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data=DEMO_CONFIG_DATA,
+        unique_id=DEMO_CONFIG_DATA[CONF_DEVICE_API_ID],
+    )
+
+
+@pytest.fixture(name="setup_entry_mock")
```

Comment:
do you even need this at all? It seems to work without it

Reply:
As long as get_device is patched it should work without it. Actually, I though it is used as a sort of shortcut that prevents calling the actual async_setup_entry. As it was included in every config_flow test I checked, I assumed that is the way to do it. Do you recommend removing it?

Reply:
My approach on this, is to remove unused code. Since you're already patching the get_device everywhere, I'd remove it.

Reply:
Actually I am not sure where it gets called. I will check whether it is called at all. Sure, if it is not called I remove it. I just assumed that it is used to prevent actually setting up the new domain. Because if it gets setup it should be deinititilized as well, right?

Reply:
no need to care about this, if you were to set it up multiple times in a single test it would crash anyways

Reply:
it gets called, but following your suggestion I removed it anyway